internal/controller/grafanaorganization_controller.go

```diff
@@ -132,3 +190,83 @@ func (r *GrafanaOrganizationReconciler) SetupWithManager(mgr ctrl.Manager) error
 		For(&v1alpha1.GrafanaOrganization{}).
 		Complete(r)
 }
+
+func (r GrafanaOrganizationReconciler) configureOrgMapping(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	organizations := v1alpha1.GrafanaOrganizationList{}
+	err := r.Client.List(ctx, &organizations)
+	if err != nil {
+		logger.Error(err, "failed to list grafana organizations.")
+		return errors.WithStack(err)
+	}
+
+	grafanaConfig := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "grafana-user-values",
+			Namespace: "giantswarm",
+		},
+		Data: map[string]string{},
+	}
+
+	_, err = controllerutil.CreateOrPatch(ctx, r.Client, &grafanaConfig, func() error {
+		var sb strings.Builder
+		sb.WriteString(fmt.Sprintf("\"*:%s:%s\"", sharedOrgName, grafanaAdminRole))
+		for _, organization := range organizations.Items {
+			rbac := organization.Spec.RBAC
+			organizationName := organization.Spec.DisplayName
+			for _, adminOrgAttribute := range rbac.Admins {
+				buildOrgMapping(&sb, organizationName, adminOrgAttribute, grafanaAdminRole)
+			}
+			for _, editorOrgAttribute := range rbac.Editors {
+				buildOrgMapping(&sb, organizationName, editorOrgAttribute, grafanaEditorRole)
+			}
+			for _, viewerOrgAttribute := range rbac.Viewers {
+				buildOrgMapping(&sb, organizationName, viewerOrgAttribute, grafanaViewerRole)
+			}
+			// Set owner reference to the config map to be able to clean it up when all organizations are deleted
+			err = controllerutil.SetOwnerReference(&organization, &grafanaConfig, r.Scheme)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+		}
+
+		orgMapping := sb.String()
+		logger.Info("configuring org mapping", "orgMapping", orgMapping)
+
+		data := struct {
+			OrgMapping string
+		}{
+			OrgMapping: orgMapping,
+		}
+
+		var values bytes.Buffer
+		err = grafanaUserConfigTemplate.Execute(&values, data)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		grafanaConfig.Data = make(map[string]string)
+		grafanaConfig.Data["values"] = values.String()
+
+		return nil
+	})
+	if err != nil {
+		logger.Error(err, "failed to generate grafana user configmap values.")
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func buildOrgMapping(sb *strings.Builder, organizationName, userOrgAttribute, role string) {
+	// We preprend with a space and we add double quotes to the org mapping to support spaces in display names
+	sb.WriteString(" \"")
+	// We need to escape the colon in the userOrgAttribute
+	sb.WriteString(strings.ReplaceAll(userOrgAttribute, ":", "\\:"))
+	sb.WriteString(":")
+	sb.WriteString(organizationName)
+	sb.WriteString(":")
+	sb.WriteString(role)
+	sb.WriteString("\"")
+}
```

Comment:
Please use a slice and strings.Join with comma rather than building the `org_mapping` using a string.Builder as this would lead to trailing spaces/comma.

Reply:
We're in a loop, slice.join is readable but this will cause really pour performance

Reply:
Also, this does not add any commas or spaces, i'm confused

Reply:
This is very hard to read a string slice would make things easier to maintain that's my main concern.